File: Android/Pittsburgh_Realtime_Tracker/app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/world/jsonpojo/Ptr.java

package rectangledbmi.com.pittsburghrealtimetracker.world.jsonpojo;

import java.util.ArrayList;
import java.util.List;
import javax.annotation.Generated;
import com.google.gson.annotations.Expose;

@Generated("org.jsonschema2pojo")
public class Ptr {

    @Expose
    private int pid;
    @Expose
    private double ln;
    @Expose
    private String rtdir;
    @Expose
    private List<Pt> pt = new ArrayList<Pt>();
    @Expose
    private String msg;

    /**
     *
     * @return
     *     The pid
     */
    public int getPid() {
        return pid;
    }

    /**
     *
     * @param pid
     *     The pid
     */
    public void setPid(int pid) {
        this.pid = pid;
    }

    /**
     *
     * @return
     *     The ln
     */
    public double getLn() {
        return ln;
    }

    /**
     *
     * @param ln
     *     The ln
     */
    public void setLn(double ln) {
        this.ln = ln;
    }

    /**
     *
     * @return
     *     The rtdir
     */
    public String getRtdir() {
        return rtdir;
    }

    /**
     *
     * @param rtdir
     *     The rtdir
     */
    public void setRtdir(String rtdir) {
        this.rtdir = rtdir;
    }

    /**
     *
     * @return
     *     The pt
     */
    public List<Pt> getPt() {
        return pt;
    }

    /**
     *
     * @param pt
     *     The pt
     */
    public void setPt(List<Pt> pt) {
        this.pt = pt;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

}

File: pat-static/src/main/java/com/rectanglel/patstatic/patterns/response/Ptr.kt
package com.rectanglel.patstatic.patterns.response

import java.util.ArrayList
import javax.annotation.Generated
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Whole pattern Retrofit POJO that contains all its points
 *
 * @author Jeremy Jao
 * @since 46
 */
@Generated("org.jsonschema2pojo")
data class Ptr (

    @Expose
    val pid: Int = 0,
    @Expose
    val ln: Double = 0.toDouble(),
    @Expose
    val rtdir: String,
    @SerializedName("pt")
    val pt: List<Pt>? = ArrayList(),
    @Expose
    val msg: String? = null

    ){


//    val pt: List<Pt>?
//        get() {
//            if (rtdir != null && points != null) {
//                for (p in points) {
//                    p.rtdir = rtdir
//                }
//            }
//            return points
//        }
}